wsdl/ver10/schema/onvif.xsd

```diff
@@ -7631,6 +7631,121 @@ and sample rate. </xs:documentation>
 		<xs:anyAttribute processContents="lax"/>
 	</xs:complexType>
 	<!--===============================-->
+	<xs:simpleType name="TargetFormat">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="MP4">
+				<xs:annotation>
+					<xs:documentation>
+						MP4 files with all tracks in a single file.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="CMAF">
+				<xs:annotation>
+					<xs:documentation>
+						CMAF compliant MP4 files with 1 track per file.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+		</xs:restriction>
+	</xs:simpleType>
+	<!--===============================-->
+	<xs:simpleType name="EncryptionMode">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="CENC">
+				<xs:annotation>
+					<xs:documentation>
+						Basic symmetric payload encryption as specified in ISO/IEC 23001-7.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+		</xs:restriction>
+	</xs:simpleType>
+	<!--===============================-->
+	<xs:complexType name="RecordingEncryption">
+		<xs:sequence>
+			<xs:element name="KID" type="xs:hexBinary">
+				<xs:annotation>
+					<xs:documentation>Key ID of the associated key for encryption.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="Key" type="xs:hexBinary" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>
+						Key for encrypting content.
+						This parameter must not be included when reading.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="Track" type="xs:string" minOccurs="0" maxOccurs="unbounded">
+				<xs:annotation>
+					<xs:documentation>
+						Optional list of track tokens to be encrypted.
+						If no track tokens are specified, all tracks are encrypted.
+						Each track must only be contained in one encryption configuration.
+						It is allowed to specify different encryption configurations for different tracks.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:any namespace="##any" processContents="lax" minOccurs="0" maxOccurs="unbounded"/>   <!-- first ONVIF then Vendor -->
+		</xs:sequence>
+		<xs:attribute name="Mode" type="xs:string" use="required">
+			<xs:annotation>
+				<xs:documentation>
+					Mode of encryption.
+					See tt:EncryptionMode for a list of definitions.
+				</xs:documentation>
+			</xs:annotation>
+		</xs:attribute>
+		<xs:anyAttribute processContents="lax"/>
```

Comment:
Add KeyManagmentMode attribute to be able to add a more advanced key managment in the future, e.g. using a key server

Reply:
As discussed, we will add something like this in the future and make it optional.